Build/scripts/finalize_build.ps1

```diff
@@ -26,29 +22,96 @@ if (Test-Path $PogoFolder) {
+$testSummaryFile = Join-Path $testLogsDropPath "summary.${Env:BuildName}.log"
 
-# If build_{}{}.err contains any text then there were build errors and we record that the build failed.
-$BuildFailed = $false
-if (Test-Path $buildFlavorErrFile) {
-    $BuildFailed = (Get-Item $buildFlavorErrFile).length -gt 0
+# if build.*.err contains any text then there were build errors
+$BuildSucceeded = $true
+if (Test-Path $buildErrFile) {
+    # build errors file has non-zero length iff the build failed
+    $BuildSucceeded = (Get-Item $buildErrFile).length -eq 0
 }
 
-$status = "passed"
-if ($BuildFailed) {
-    $status = "failed"
+# if summary.*.err contains any text then there were test failures
+$TestsPassed = $true
+if (Test-Path $testSummaryFile) {
+    # summary file has non-zero length iff the tests failed
+    $TestsPassed = (Get-Item $testSummaryFile).length -eq 0
 }
 
-$buildFlavorJsonFile = Join-Path $Env:DropPath "${Env:BuildType}.json"
+$Status = "passed"
+if ((-not $BuildSucceeded) -or (-not $TestsPassed)) {
+    $Status = "failed"
+}
+
+$buildFlavorJsonFile = Join-Path $Env:DropPath "${Env:BuildName}.json"
 $buildFlavorJson = New-Object System.Object
 
-$buildFlavorJson | Add-Member -type NoteProperty -name status -value $status
+$buildFlavorJson | Add-Member -type NoteProperty -name buildName -value $Env:BuildName
+$buildFlavorJson | Add-Member -type NoteProperty -name status -value $Status
+$buildFlavorJson | Add-Member -type NoteProperty -name BuildSucceeded -value $BuildSucceeded
+$buildFlavorJson | Add-Member -type NoteProperty -name testsPassed -value $TestsPassed
 $buildFlavorJson | Add-Member -type NoteProperty -name arch -value $Env:BuildPlatform
 $buildFlavorJson | Add-Member -type NoteProperty -name flavor -value $Env:BuildConfiguration
+$buildFlavorJson | Add-Member -type NoteProperty -name subtype -value $Env:BuildSubtype
 
 $buildFlavorJson | ConvertTo-Json | Write-Output
 $buildFlavorJson | ConvertTo-Json | Out-File $buildFlavorJsonFile -Encoding ascii
+
+#
+# Copy outputs to metadata directory
+#
+
+$metadataDir = Join-Path $coreSourcesDir "metadata"
+New-Item -Force -ItemType Directory -Path $metadataDir
+
+Copy-Item -Verbose -Force (Join-Path $sourcesDir "ComputedEnvironment.cmd") $metadataDir
+Copy-Item -Verbose -Force (Join-Path $coreSourcesDir "Build\scripts\compose_build.ps1") $metadataDir
+Copy-Item -Verbose -Force (Join-Path $Env:BinariesDirectory "change.json") $metadataDir
+Copy-Item -Verbose -Force (Join-Path $Env:BinariesDirectory "change.txt") $metadataDir
+Copy-Item -Verbose -Force $buildFlavorJsonFile $metadataDir
+
+# Search for *.nuspec files and copy them to $metadataDir
+Get-ChildItem -Path (Join-Path $sourcesDir "Build") "*.nuspec" `
+    | % { Copy-Item -Verbose -Force $_.FullName $metadataDir }
+
+#
+# Copy POGO directory if present for this build
+#
+
+# TODO (doilij) REMOVE speculative POGO copy since all builds will be explicitly specified now
```

Comment:
Note to self: This is not exactly true. We should keep this block because it doesn't hurt.